connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/WebhookResponseManager.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.runtime.inbound.webhook;
+
+import static io.camunda.connector.impl.Constants.RESPONSE_BODY_EXPRESSION;
+
+import io.camunda.connector.api.inbound.InboundConnectorResult;
+import io.camunda.connector.impl.ConnectorInputException;
+import io.camunda.connector.impl.inbound.InboundConnectorProperties;
+import io.camunda.connector.runtime.core.feel.FeelEngineWrapper;
+import io.camunda.connector.runtime.core.feel.FeelEngineWrapperException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+
+public class WebhookResponseManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(WebhookResponseManager.class);
+
+  public static final String PROCESS_DATA_KEY = "processData";
+
+  private final FeelEngineWrapper feelEngine;
+
+  @Autowired
+  public WebhookResponseManager(final FeelEngineWrapper feelEngine) {
+    this.feelEngine = feelEngine;
+  }
+
+  public ResponseEntity produceResponse(
+      final InboundConnectorResult correlatedResult,
+      final InboundConnectorProperties properties,
+      final Object context) {
+    Map evaluatedResponseBody =
+        Optional.ofNullable(evaluateResponseBodyExpression(properties, context))
+            .orElse(Collections.emptyMap());
+    Map response = new HashMap(evaluatedResponseBody);
+    response.put(PROCESS_DATA_KEY, correlatedResult);
```

Comment:
I thought the idea was to allow users to decide whether they want to include the `correlatedResult` into the response or not based on their `responseExpression`. We dont need to add it here. Maybe only if the `responseExpression` is null.

Reply:
Alright, that makes sense. Changed and added some unit tests.